src/convertToObject.js

```diff
@@ -6,7 +6,15 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  return sourceString.split(';').reduce((accumulator, element) => {
+    const [key, value] = element.split(':').map((part) => part.trim());
+
+    if (key && value) {
```

Comment:
The current check `if (key && value)` might not handle all edge cases, such as when `key` or `value` is an empty string or contains only whitespace. Consider refining this condition to ensure that only valid key-value pairs are added to the accumulator.